lib/mail_room/mailbox_watcher.rb

```diff
@@ -104,7 +116,9 @@ def stop_idling
       return unless idling?
 
       imap.idle_done
+      
       idling_thread.join
+      self.idling_thread = nil
```

Comment:
I don't think we do, since that's in the ensure section for #idle, which is where we get after idle_done is called and imap.idle returns.